cmd/utils/cmd.go

```diff
@@ -211,8 +215,11 @@ func ImportChain(chain *core.BlockChain, fn string) error {
 			log.Info("Skipping batch as all blocks present", "batch", batch, "first", blocks[0].Hash(), "last", blocks[i-1].Hash())
 			continue
 		}
-		if _, err := chain.InsertChain(missing); err != nil {
-			return fmt.Errorf("invalid block %d: %v", n, err)
+		if failindex, err := chain.InsertChain(missing); err != nil {
```

Comment:
I think we can simplify it a bit.

```
diff --git a/cmd/utils/cmd.go b/cmd/utils/cmd.go
index e1bafc53c3..37b6d222a5 100644
--- a/cmd/utils/cmd.go
+++ b/cmd/utils/cmd.go
@@ -211,8 +211,14 @@ func ImportChain(chain *core.BlockChain, fn string) error {
                        log.Info("Skipping batch as all blocks present", "batch", batch, "first", blocks[0].Hash(), "last", blocks[i-1].Hash())
                        continue
                }
-               if _, err := chain.InsertChain(missing); err != nil {
-                       return fmt.Errorf("invalid block %d: %v", n, err)
+               if failIndex, err := chain.InsertChain(missing); err != nil {
+                       var failNumber uint64
+                       if failIndex == -1 {
+                               failNumber = missing[0].NumberU64()
+                       } else {
+                               failNumber = missing[failIndex].NumberU64()
+                       }
+                       return fmt.Errorf("invalid block %d: %v", failNumber, err)
                }
        }
```

Reply:
I'd change @rjl493456442 's condition-check into 


```golang
+                       if failIndex > 0 && failIndex < len(missing) {
+                               failNumber = missing[failIndex].NumberU64()
+                       } else {
+                               failNumber = missing[0].NumberU64()
+                       }
```